diff --git a/src/jloptions.c b/src/jloptions.c
index 4b11093..04f6f42 100644
--- a/src/jloptions.c
+++ b/src/jloptions.c
@@ -620,8 +620,9 @@ restart_switch:
     }
     jl_options.code_coverage = codecov;
     jl_options.malloc_log = malloclog;
-    *argvp += optind;
-    *argcp -= optind;
+    int proc_args = *argcp < optind ? *argcp : optind;
+    *argvp += proc_args;
+    *argcp -= proc_args;
 }
 
 JL_DLLEXPORT void jl_set_ARGS(int argc, char **argv)
